test/unit/test-mysqlQuery.js

```diff
@@ -94,6 +105,17 @@ describe('MysqlQuery', function(){
             }
             Should.not.exist(error);
         });
+        it('Should throw an error for INSERT ... ON DUPLICATE KEY UPDATE queries', function() {
```

Comment:
I think this is the test that needs to get removed along with the check on insert queries.